ext/civi_case/Civi/Afform/Behavior/ContactAutofillBasedOnCase.php

```diff
@@ -0,0 +1,212 @@
+<?php
+namespace Civi\Afform\Behavior;
+
+use Civi\Afform\Event\AfformEntitySortEvent;
+use Civi\Afform\Event\AfformPrefillEvent;
+use Civi\API\Event\RespondEvent;
+use Civi\Api4\Utils\CoreUtil;
+use Civi\Core\Event\GenericHookEvent;
+use Civi\Core\Service\AutoService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use CRM_Case_ExtensionUtil as E;
+
+/**
+ * @service
+ * @internal
+ */
+class ContactAutofillBasedOnCase extends AutoService implements EventSubscriberInterface {
+
+  /**
+   * @return array
+   */
+  public static function getSubscribedEvents() {
+    return [
+      'civi.api.respond' => 'onAfformGetBehavors',
+      'hook_civicrm_angularModules' => 'onAngularModules',
+      'hook_civicrm_buildAsset' => 'onBuildAsset',
+      'civi.afform.sort.prefill' => 'onAfformSortPrefill',
+      'civi.afform.prefill' => ['onAfformPrefill', 100],
+    ];
+  }
+
+  public static function onAfformGetBehavors(RespondEvent $event) {
+    if ($event->getEntityName() == 'AfformBehavior' && $event->getActionName() == 'get') {
+      $response = $event->getResponse();
+      foreach ($response as $index => $behavior) {
+        foreach (\CRM_Contact_BAO_ContactType::basicTypes() as $contactType) {
+          if (in_array($contactType, $behavior['entities'])) {
+            $response[$index]['modes'][$contactType] = self::getCaseRoleModes($contactType, $behavior['modes'][$contactType]);
+          }
+        }
+      }
+      $event->setResponse($response);
+    }
+  }
+
+  public static function onAngularModules(GenericHookEvent $event) {
```

Comment:
I don't think this hook is needed if you enable the `ang-php@1.0.0` mixin.

Reply:
I have added a comment to explain this

Reply:
Another way to achieve this @jaapjansma would be to set `basePages` in your new `.ang.php` file so it will always be loaded on the admin afform page, like this:

https://github.com/civicrm/civicrm-core/blob/490565d08a13dbd2fd2a7887f29ff9c5fe2f187c/ext/afform/admin/ang/afAdmin.ang.php#L13

Reply:
Ah I did not know that. I have updated the PR so it will use the `basePages` instead.